### PR TITLE
Drivable area

### PR DESCRIPTION
**Pull request by XiaLiPKU**
add the calss `background` for drivable area task.
Fix the problem that out_paths and color_lists are of different length (cause by a `continue`)

### CODE REVIEW

File: doc/source/format.rst [hunk 1]
```diff
@@ -45,35 +45,84 @@ Semantic Segmentation
...
+    11: person
+    12: rider
+    13: car
+    14: truck
+    15: bus
+    16: train
+    17: motorcycle
+    18: bicycle
+
+`category_id` ranges from **0** for the semantic segmentation task.
```
Comment (XiaLiPKU) [inline doc/source/format.rst:68]: "from -1" error fixed here

File: bdd100k/label/to_mask.py [hunk 2]
```diff
@@ -172,18 +211,40 @@ def semseg_to_masks(
...
+semseg_to_masks: ToMasksFunc = partial(
+    seg_to_masks, mode="sem_seg", back_color=IGNORE_LABEL, closed=True
+)
+drivable_to_masks: ToMasksFunc = partial(
+    seg_to_masks,
+    mode="drivable",
+    back_color=len(drivables) - 1,
+    closed=True,
+)
+lanemark_to_masks: ToMasksFunc = partial(
```
Comment (XiaLiPKU) [inline bdd100k/label/to_mask.py:243]: lane_mark to mask draw non-closed polygons.
Other tasks to_mask functions draw closed polygons

File: bdd100k/label/to_mask.py [hunk 3]
```diff
@@ -172,18 +211,40 @@ def semseg_to_masks(
...
 
 
+ToMasksFunc = Callable[[List[Frame], str, bool, bool, int], None]
+semseg_to_masks: ToMasksFunc = partial(
+    seg_to_masks, mode="sem_seg", back_color=IGNORE_LABEL, closed=True
+)
+drivable_to_masks: ToMasksFunc = partial(
+    seg_to_masks,
+    mode="drivable",
+    back_color=len(drivables) - 1,
```
Comment (XiaLiPKU) [inline bdd100k/label/to_mask.py:240]: For drivable, the default color (for background) is set as 2 here.
For sem_seg and drivable, this value is still 255 (IGNORE_LABEL)

File: bdd100k/label/to_mask.py [hunk 4]
```diff
@@ -33,22 +33,35 @@
...
     poly2ds: List[List[Poly2D]],
     with_instances: bool = True,
+    back_color: int = 0,
+    closed: bool = True,
 ) -> None:
     """Converting a frame of poly2ds to mask/bitmask."""
     assert len(colors) == len(poly2ds)
 
+    if with_instances:
+        img = np.ones([*SHAPE, 4], dtype=np.uint8) * back_color
```
Comment (XiaLiPKU) [inline bdd100k/label/to_mask.py:57]: RGBA bitmask for ins_seg, seg_track

File: bdd100k/label/to_mask.py [hunk 5]
```diff
@@ -33,22 +33,35 @@
...
+    else:
+        img = np.ones([*SHAPE, 1], dtype=np.uint8) * back_color
```
Comment (XiaLiPKU) [inline bdd100k/label/to_mask.py:59]: one-channel mask for seg_track, drivable and lane_marking

**Review by fyu — APPROVED** [review]

**Pull request merged**